fdbclient/tests/mocks3_fixture.sh

```diff
@@ -0,0 +1,145 @@
+#!/bin/bash
+
+# Mock S3 Server fixture for ctests
+# Replaces SeaweedFS with in-process MockS3Server
+
+set -e
+
+MOCKS3_HOST=${MOCKS3_HOST:-127.0.0.1}
+MOCKS3_PORT=${MOCKS3_PORT:-8080}
+MOCKS3_PID=""
+
+# Find available port by actually trying to start the server
+find_available_port() {
+    local start_port=$1
+    local max_attempts=10
+    local port=$start_port
+    
+    for attempt in $(seq 1 $max_attempts); do
+        # Check if port is available using netstat (faster than starting server)
+        if ! netstat -tuln 2>/dev/null | grep -q ":$port "; then
+            echo $port
+            return 0
+        fi
+        
+        # Try next port
+        port=$((port + 1))
+    done
+    
+    echo "ERROR: Could not find available port after $max_attempts attempts" >&2
+    return 1
+}
+
+start_mocks3() {
+    local build_dir_param="${1:-}"
+    local max_attempts=10
+    local attempt
+    
+    # Use provided build directory or try to find it
+    BUILD_DIR="$build_dir_param"
+    if [ -z "$BUILD_DIR" ]; then
+        # Find the build directory - look for fdbserver binary
+        for dir in "../build_output" "../../build_output" "../../../build_output" "./build_output" "."; do
+            if [ -f "$dir/bin/fdbserver" ]; then
+                BUILD_DIR="$dir"
+                break
+            fi
+        done
+    fi
+
+    if [ -z "$BUILD_DIR" ]; then
+        echo "ERROR: Could not find fdbserver binary to run MockS3Server"
+        return 1
+    fi
+    
+    if [ ! -f "$BUILD_DIR/bin/fdbserver" ]; then
```

Comment:
```suggestion
    if [ ! -x "$BUILD_DIR/bin/fdbserver" ]; then
```

I assume we want to execute the `fdbserver` binary?